test/e2e/scaffold/scaffold.go

```diff
@@ -175,6 +179,17 @@ func (s *Scaffold) NewAPISIXHttpsClient() *httpexpect.Expect {
 					// accept any certificate; for testing only!
 					InsecureSkipVerify: true,
 				},
+				DialContext: func(ctx context.Context, network, addr string) (net.Conn, error) {
```

Comment:
Why not set the `ServerName` in `tls.Config`?

Reply:
Great, use `ServerName` instead.